getters.go

```diff
@@ -36,36 +36,23 @@ func newParsedFile(file *ast.File, fset *token.FileSet) (*parsedFile, error) {
 		file: file,
 	}
 
-	var err error
-
 	// Read original file. This is necessary for making a replacements for
 	// inline comments. I couldn't find a better way to get original line
 	// with code and comment without reading the file. Function `Format`
 	// from "go/format" won't help here if the original file is not gofmt-ed.
-	pf.lines, err = readFile(file, fset)
-	if err != nil {
-		return nil, fmt.Errorf("read file: %w", err)
-	}
 
-	// Dirty hack. For some cases Go generates temporary files during
-	// compilation process if there is a cgo block in the source file. Some of
-	// these temporary files are just copies of original source files but with
-	// new generated comments at the top. Because of them the content differs
-	// from AST. For some reason it differs only in golangci-lint. I failed to
-	// find out the exact description of the process, so let's just skip files
-	// generated by cgo.
-	if isCgoGenerated(pf.lines) {
-		return nil, errUnsuitableInput
+	var filename string
+	filename, _ = getFilename(fset, file)
+
+	if !strings.HasSuffix(filename, ".go") {
```

Comment:
That's already checked in `getFilename`, right?

Reply:
No, in `getFilename` I check only the filename at the adjusted position.

Here I evaluate the filename at the adjusted position and the non-adjusted position.